api/grpcserver/mesh_service.go

```diff
@@ -610,3 +614,59 @@ func (s MeshService) EpochStream(req *pb.EpochStreamRequest, stream pb.MeshServi
 	)
 	return nil
 }
+
+func (s MeshService) MalfeasanceQuery(ctx context.Context, req *pb.MalfeasanceRequest) (*pb.MalfeasanceResponse, error) {
+	parsed, err := hex.DecodeString(req.SmesherHex)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+	id := types.BytesToNodeID(parsed)
```

Comment:
I think you still need to check the length of `parsed` otherwise missing bytes will be filled with zeroes and extra bytes silently dropped.